GCC -15 warnings

/src/H5B.c:1437:46: warning: potential null pointer dereference [-Wnull-dereference]
 1437 |     if (*lt_key_changed && H5_addr_defined(bt->left)) {
src/H5private.h:440:36: note: in definition of macro ‘H5_addr_defined’
  440 | #define H5_addr_defined(X)       ((X) != HADDR_UNDEF)
      |                                    ^
src/H5B.c:1452:51: warning: potential null pointer dereference [-Wnull-dereference]
 1452 |     else if (*rt_key_changed && H5_addr_defined(bt->right)) {
src/H5private.h:440:36: note: in definition of macro ‘H5_addr_defined’
  440 | #define H5_addr_defined(X)       ((X) != HADDR_UNDEF)

--- a/src/H5B.c
+++ b/src/H5B.c
@@ -1434,7 +1434,7 @@ H5B__remove_helper(H5F_t *f, haddr_t addr, const H5B_class_t *type, int level, u
         ret_value = H5B_INS_NOOP;
 
     /* Patch keys in neighboring trees if necessary */
-    if (*lt_key_changed && H5_addr_defined(bt->left)) {
+    if (*lt_key_changed && bt != NULL && H5_addr_defined(bt->left)) {
         assert(type->critical_key == H5B_LEFT);
         assert(level > 0);
 
@@ -1449,7 +1449,7 @@ H5B__remove_helper(H5F_t *f, haddr_t addr, const H5B_class_t *type, int level, u
             HGOTO_ERROR(H5E_BTREE, H5E_CANTUNPROTECT, H5B_INS_ERROR, "unable to release node from tree");
         sibling = NULL; /* Make certain future references will be caught */
     }                   /* end if */
-    else if (*rt_key_changed && H5_addr_defined(bt->right)) {
+    else if (*rt_key_changed && bt != NULL && H5_addr_defined(bt->right)) {
         assert(type->critical_key == H5B_RIGHT);
         assert(level > 0);
 

